### PR TITLE
demux_lavf: pass jpg filenames to ffmpeg for probing

### LINKED ISSUE #13192: mpv shows black & white image instead of original JPEG
### Important Information

- mpv version
0.36.0
- Linux Distribution and Version
Arch Linux x86-64 up to date
- Source of the mpv binary
https://archlinux.org/packages/extra/x86_64/mpv/

### Reproduction steps

I am using mpv as an image viewer.
Run `mpv --no-config --image-display-duration=inf --keep-open i.jpg` with the provided file.

### Expected behavior

The original colored image is shown. 

### Actual behavior

A white image is shown, for other images it is a black and white version of the image. It is easily reproducible with images generated by different smartphones.

It seems to be caused by some metadata embedded in the image, because if I run `jpegoptim --strip-all xxx.jpg` on the files, the issue does not happen, same if I extract the image stream and pipe it to mpv with `ffmpeg -i xxx.jpg -map v:1 -c:v copy -f matroska - | mpv -`

### Log file

https://x0.at/mnkH.log

### Sample files

Test image: https://x0.at/1Jrk.jpg

### LINKED ISSUE #13431: [BUG] Specific image file fails to open
### Important Information

- mpv v0.37.0-241-gfe0c181b
- Windows 10 Home 22H2 19045.3930
- [Source of the mpv binary](https://sourceforge.net/projects/mpv-player-windows/files/)
- NVIDIA GeForce GTX 1650 4GB VRAM | Driver: 551.23

### Reproduction steps
- Try to open the image file "42974757_p0.jpg" (848kb) by any method: terminal, drag and drop, playlist-next/prev, etc.
- File fails to load, and next file in directory is loaded isntead (I use autoload.lua)
### Expected behavior
- Image loads normally, like any other
### Actual behavior
- Image fails to load; it's the first and only one I ever found this issue
### Log file
[output.txt](https://github.com/mpv-player/mpv/files/14188632/output.txt)
[output no-config no-scripts.txt](https://github.com/mpv-player/mpv/files/14188619/output.no-config.no-scripts.txt)

### Sample files
42974757_p0
Must be this exact file (with 848kb). When I re-downloaded it from the source (Pixiv) or uploaded it to Imgur, the bug stopped to occur. Maybe the file is simply corrupted, but it opened normally on ImageGlass and Photoshop.

### PR DESCRIPTION
**Pull request by guidocella**
Fixes #13431, fixes #13192. See the comment for details.

### CODE REVIEW

**Comment by guidocella** [comment]
It's surely better to fix ffmpeg's parsing but that is much harder to do so this fixes our issues for the time being, as images without extensions are rare in practice and smartphone photos not working is really annoying.

**Comment by guidocella** [comment]
I never saw mpv misdetect mp3s and 41243e7c4f doesn't provide example files either, it may even have been fixed since then, so I think we should only add mp3 after someone actually needs it for some file.

Enabling it for all files can prevent discovering the right codec for files with a wrong extension, just doing `mv foo.jpg foo.png` and `mpv foo.png` errors with `[ffmpeg/video] png: Invalid PNG signature 0xFFD8FFDB00840008`, while it detects it as jpg without passing `priv->filename`.

Just passing jpg filenames doesn't re-enable image2 as long is it's blacklisted. I already tried putting one of those photos on a webserver as `foo%d.jpg` and `mpv http://example.com/foo%d.jpg` or `mpv http://example.com/foo%25d.jpg` only try to play that file, you have to explicitly do `mpv --demuxer-lavf-format=image2 http://example.com/foo%d.jpg` for ffmpeg to interpret it as a pattern; it logs:

```
[ffmpeg] http: HTTP error 404 Not Found
[ffmpeg] http: HTTP error 404 Not Found
[ffmpeg] http: HTTP error 404 Not Found
[ffmpeg] http: HTTP error 404 Not Found
[ffmpeg] http: HTTP error 404 Not Found
[ffmpeg/demuxer] image2: Could find no file with path 'http://example.com/foo%d.jpg' and index in the range 0-4
```

(you can also test this locally but it's easier to understand what's happening with 404 errors being logged)

**Comment by guidocella** [comment]
> > just doing `mv foo.jpg foo.png` and `mpv foo.png` errors
>
> User error.

Well according to 74e62ed2d1 there are many such files in the wild (but I can't verify this claim), and while ffplay still plays foo.png there is no public API to replicate this without enabling image2 and thus pattern matching (actually mpv with image2 just segfaults in this case). I'm also not aware of filetypes other than jpg and m3u that currently need the filename.

**Review by kasper93 — APPROVED** [review]
Ok, we can go with this. I think it won't make more harm.

File: demux/demux_lavf.c [hunk 1]
```diff
@@ -464,12 +464,17 @@ static int lavf_check_file(demuxer_t *demuxer, enum demux_check check)
         }
     }
 
-    // HLS streams seems to be not well tagged, so matching mime type is not
-    // enough. Strip URL parameters and match extension.
-    bstr ext = bstr_get_ext(bstr_split(bstr0(priv->filename), "?#", NULL));
+    // HLS streams do not seem to be well tagged, so matching by MIME type is
+    // not enough - we need to strip the query string and match by their
+    // extensions. We also pass jpg filenames to fix issues like #13192 and
+    // #13431. We don't pass filenames otherwise to not misdetect files with a
```
[the remark this answers is not in the document]
Comment (guidocella) [reply demux/demux_lavf.c]: Added a summary of the issues.

**Comment by kasper93** [comment]
We will monitor how big of a problem it is. At least other image formats works as .jpg. I wouldn't worry about mp3 with .jpg extension is not that common.

**Pull request merged**